packages/react-native-platform-override/src/Manifest.ts

```diff
@@ -5,183 +5,82 @@
  * @format
  */
 
-import * as ManifestData from './ManifestData';
+import * as Serialized from './Serialized';
 
 import * as _ from 'lodash';
-import * as crypto from 'crypto';
-import * as path from 'path';
 
+import Override, {deserializeOverride} from './Override';
 import {OverrideFileRepository, ReactFileRepository} from './FileRepository';
+import OverrideFactory from './OverrideFactory';
+import {ValidationError} from './ValidationStrategy';
 
-export interface ValidationError {
-  type:
-    | 'fileMissingFromManifest' // An override file is present with no manifest entry
-    | 'overrideFileNotFound' // The manifest describes a file which does not exist
-    | 'baseFileNotFound' // The base file for a manifest entry cannot be found
-    | 'outOfDate'; // A base file has changed since the manifested version
-  file: string;
-}
-
-export type OverrideType = 'platform' | 'patch' | 'derived' | 'copy';
-
-interface OverrideBaseInfo {
-  baseFile: string;
-  baseVersion: string;
-  baseHash: string;
-}
-
+/**
+ * Represents a collection of overrides listed in an on-disk manifest. Allows
+ * performing aggregate operations on the overrides.
+ */
 export default class Manifest {
-  private overrides: Array<ManifestData.Entry>;
-  private overrideRepo: OverrideFileRepository;
-  private reactRepo: ReactFileRepository;
+  private overrides: Array<Override>;
 
-  constructor(
-    data: ManifestData.Manifest,
-    overrideRepo: OverrideFileRepository,
-    reactRepo: ReactFileRepository,
-  ) {
-    this.overrides = _.cloneDeep(data.overrides);
-    this.overrideRepo = overrideRepo;
-    this.reactRepo = reactRepo;
+  constructor(overrides: Array<Override>) {
+    this.overrides = _.cloneDeep(overrides);
```

Comment:
We should validate we don't have duplicates here and make sure to propagate that error up further.